components/Operations.js

```diff
@@ -13,36 +13,39 @@ import { FormatHelpers } from '../helpers/format';
 
 export function Operations({ asyncapi }) {
   const channels = asyncapi.channels();
-  if (!Object.keys(channels).length) {
+  if (channels.isEmpty()) {
     return null;
   }
 
   const operationsList = [];
-  Object.entries(channels).forEach(([channelName, channel]) => {
-    if (channel.hasPublish()) {
-      operationsList.push(
-        <Operation
-          key={`pub-${channelName}`}
-          type='publish'
-          asyncapi={asyncapi}
-          operation={channel.publish()}
-          channelName={channelName}
-          channel={channel}
-        />
-      );
-    }
-    if (channel.hasSubscribe()) {
-      operationsList.push(
-        <Operation
-          key={`sub-${channelName}`}
-          type='subscribe'
-          asyncapi={asyncapi}
-          operation={channel.subscribe()}
-          channelName={channelName}
-          channel={channel}
-        />
-      );
-    }
+  channels.all().map(channel => {
+    const channelName = channel.address();
+    channel.operations().all().forEach(operation => {
+      if (operation.action() === 'publish') {
```

Comment:
shouldn't we use `isSend` and `isReceive` instead of hardcoding to `publish` and `subscribe` that soon go away? Also this is the point of `is*` helpers right?

Reply:
I plan on doing a followup that clears these things up 🙂

Reply:
you mean a separate PR?

Reply:
Yes 🙂